### DISCUSSION TITLE
prompt validator

### DISCUSSION THREAD

**Discussion by wiiiktor** [guidance-ai/guidance#487] [General]
Is there any prompt validator or plans to make one? 
I mean a function that checks, if prompt has a valid Guidance syntax (let's way I used LLM to create a Guidance prompt and before I use it, I want to make sure it has a valid syntax).

**Comment by slundberg** [comment]
Since guidance is now pure python based in v0.1+, this is basically asking if we have valid python code. We could in theory build a guidance function that specifies the grammar of valid guidance python programs. You could then use that as either a validator or a constraint.  ...we don't have such a function written though at the mement.
